packages/sanity/src/presentation/overlays/PostMessageDocuments.tsx

```diff
@@ -79,7 +79,17 @@ const PostMessageDocuments: FunctionComponent<PostMessageDocumentsProps> = (prop
 
   useEffect(() => {
     return comlink.on('visual-editing/fetch-snapshot', async (data) => {
-      const snapshot = await client.getDocument(data.documentId, {
+      /**
+       * This is a temporary hack to get around the fact that the `drafts.versions.versionName.id` format is not supported
+       * `versions.` don't have draft documents.
+       *
+       * Where is the drafts. prefix coming from and why it's added to the `versions.` documents?
+       * */
+      const cleanId =
+        data.documentId.includes('versions.') && data.documentId.startsWith('drafts.')
+          ? data.documentId.replace('drafts.', '')
+          : data.documentId
+      const snapshot = await client.getDocument(cleanId, {
```

Comment:
wouldn't the getPublishedId work for this? 🤔

Reply:
I don't think it would, as we're not trying to get the published id here, we're trying to get the versions id, but without the drafts prefix.
It's not clear yet how the `drafts.` prefix is able to sneak in to begin with, but my hunch it's something to do with our stega setup in `@sanity/client/stega` 🤔

Reply:
I see 🤔
Alright, in in any case I think we could still make this change? (with the right imports and all)

```suggestion
       * This is a temporary hack to get around the fact that the `drafts.versions.versionName.id` format is not supported
       * `versions.` don't have draft documents.
       *
       * Where is the drafts. prefix coming from and why it's added to the `versions.` documents?
       * */
      const cleanId =
        isVersionId(data.documentId) && isDraftId(data.documentId)
          ? data.documentId.replace('drafts.', '')
          : data.documentId
      const snapshot = await client.getDocument(cleanId, {
```

Reply:
I think it'll fail. `isVersionId` uses `id.startsWith('versions.')`, but we need it to have `id.includes('versions.')` for the `drafts.versions.` condition to be a match, and then be fixed to `versions.`?

Reply:
Ahh right, when I read through the code I didn't realise it's "includes" and not "startsWith". Right right. Nevermind me, sorry about the detour

Reply:
@pedrobonamin quick question on this: do you remember the reason why we're having this condition vs using our isDraftId / isVersionId? (specifically the latter since the condition wouldn't match) :)

Reply:
I think it was due to what @stipsan mentions, that sometimes the id here was coming as `drafts.versions` which I couldn't really track where it was coming from. 
Sometimes it was receiving: `drafts.versions.r123.foo` which is not a valid id.